src/client/app/needs/new_need_register.controller.js

```diff
@@ -48,7 +50,11 @@
         };
 
         function activate() {
-            vm.regions = CreateNeedFactory.getRegions();
+            if(!$sessionStorage.token){
+                $state.go('login');
```

Comment:
save the attempt url prior to redirecting user to the login page.
and after a successful login, redirect the user to the page he wanted, not the home page.